src/linker/Linker.Steps/MarkStep.cs

```diff
@@ -2334,6 +2376,32 @@ void MarkInterfaceImplementations (TypeDefinition type)
 				if (ShouldMarkInterfaceImplementation (type, iface))
 					MarkInterfaceImplementation (iface, new MessageOrigin (type));
 			}
+
+			bool ShouldMarkInterfaceImplementation (TypeDefinition type, InterfaceImplementation iface)
+			{
+				if (Annotations.IsMarked (iface))
+					return false;
+
+				if (!Context.IsOptimizationEnabled (CodeOptimizations.UnusedInterfaces, type))
+					return true;
+
+				if (Context.Resolve (iface.InterfaceType) is not TypeDefinition resolvedInterfaceType)
+					return false;
+
+				if (Annotations.IsMarked (resolvedInterfaceType))
+					return true;
+
+				// It's hard to know if a com or windows runtime interface will be needed from managed code alone,
+				// so as a precaution we will mark these interfaces once the type is instantiated
+				if (resolvedInterfaceType.IsImport || resolvedInterfaceType.IsWindowsRuntime)
+					return true;
+
+				if (resolvedInterfaceType.IsTypeOf<IDynamicInterfaceCastable> ())
```

Comment:
Why is this needed, should this be fixed on libraries side?

Reply:
It seems to be it'd be better to add the interface to corelib descriptors instead of hardcoding in linker

Reply:
Good point, and it turns out IDynamicInterfaceCastable is in the corelib descriptors, we just need to make sure we pass the framework libraries as 'trim' instead of 'skip'.